Can not build with `OTELCPP_MAINTAINER_MODE=ON` and examples

**Describe your environment**

Compiler: Clang 20.1.3
OS: Linux

```bash
/data/home/owentou/workspace/github/owent/opentelemetry-cpp/examples/otlp/http_instrumented_main.cc:61:12: error: unused variable 'serialize' [-Werror,-Wunused-variable]
   61 | std::mutex serialize;
      |            ^~~~~~~~~
5 warnings and 1 error generated.
```

--- a/examples/otlp/http_instrumented_main.cc
+++ b/examples/otlp/http_instrumented_main.cc
@@ -4,7 +4,7 @@
 #include <chrono>
 #include <iostream>
 #include <memory>
-#include <mutex>
+#include <mutex>  // IWYU pragma: keep
 #include <string>
 #include <utility>
 

--- a/examples/otlp/http_instrumented_main.cc
+++ b/examples/otlp/http_instrumented_main.cc
@@ -58,9 +58,9 @@
 namespace
 {
 
-std::mutex serialize;
-
 #ifdef ENABLE_THREAD_INSTRUMENTATION_PREVIEW
+
+std::mutex serialize;
 
 /**
  The purpose of MyThreadInstrumentation is to demonstrate

